src/databricks/sql/thrift_backend.py

```diff
@@ -335,15 +368,20 @@ def attempt_request(attempt):
 
             error, error_message, retry_delay = None, None, None
             try:
+
                 # The MagicMocks in our unit tests have a `name` property instead of `__name__`.
```

Comment:
I know this isn't part of this review, but I feel like something is off when I see mention to test frameworks in production code.  What is the line below doing, and how/why does it matter what MagicMocks does?

Reply:
This is excellent feedback and has been on my mind. I just opened https://github.com/databricks/databricks-sql-python/pull/188 to fix this by adding `autospec` to our Mocks and cleaning up the application logic.

The code comment you found was a result of rushing a change out the door. This is good tech debt to tidy up now.

Reply:
#188 merged and I rebased that change into this branch 👍